### PR TITLE
NPM package security audit & updates

### PR DESCRIPTION
**Pull request by ambrauer**
## Description
Updates to all packages, samples, e2e per GitHub Dependabot alerts and manual `npm audit` results. All except the Angular sample app (to be addressed in separate PR) now have either 0 or low severity (w/ blocking/breaking updates) vulnerabilities remaining.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update (non-breaking change; modified files are limited to the `/docs` directory)

## Checklist:

- [ ] I have read the Contributing guide.
- [ ] My code follows the code style of this project.
- [ ] My code/comments/docs fully adhere to the Code of Conduct.
- [ ] My change is a code change and it requires an update to the documentation.
- [ ] My change is a documentation change and it requires an update to the navigation.

### CODE REVIEW

File: samples/nextjs/package.json [hunk 1]
```diff
@@ -76,7 +76,6 @@
     "eslint-plugin-react": "^7.21.5",
     "eslint-plugin-yaml": "^0.2.0",
     "graphql-let": "^0.16.2",
-    "graphql-tag": "~2.10.1",
```
Comment (illiakovalenko) [inline samples/nextjs/package.json:79]: @ambrauer Did you run nextjs sample after removing of `graphql-tag`? I remember that it's required by graphql-let.
https://github.com/piglovesyou/graphql-let/blob/main/package.json#L140

File: samples/nextjs/package.json [hunk 2]
```diff
@@ -52,7 +52,7 @@
   "devDependencies": {
     "@graphql-codegen/cli": "^1.19.1",
     "@graphql-codegen/plugin-helpers": "^1.18.1",
-    "@graphql-codegen/typescript": "^1.17.11",
+    "@graphql-codegen/typescript": "^1.21.1",
```
Comment (illiakovalenko) [inline samples/nextjs/package.json:55]: @ambrauer Is graphql-let working fine if you update only this package?

**Review by illiakovalenko — CHANGES_REQUESTED** [review]

File: samples/nextjs/package.json (hunk 1 above)
Comment (ambrauer) [reply samples/nextjs/package.json:79]: @illiakovalenko This is actually a duplicate - we also have `graphql-tag` under `dependencies` (which was for a newer version and is still present).

File: samples/nextjs/package.json (hunk 2 above)
Comment (ambrauer) [reply samples/nextjs/package.json:55]: @illiakovalenko I'm not sure what you mean by "only"? I will double check, but graphql-let should be working fine.

File: samples/nextjs/package.json (hunk 1 above)
Comment (illiakovalenko) [reply samples/nextjs/package.json:79]: Got it

File: samples/nextjs/package.json (hunk 2 above)
Comment (illiakovalenko) [reply samples/nextjs/package.json:55]: Cool, I mean that we have another `@graphql-codegen` dependencies. If it's working - it's good.

**Review by illiakovalenko — APPROVED** [review]

**Pull request merged**